src/values/AbstractObjectValue.js

```diff
@@ -482,16 +482,16 @@ export default class AbstractObjectValue extends AbstractValue {
 
   $GetPartial(P: AbstractValue | PropertyKeyValue, Receiver: Value): Value {
     if (!(P instanceof AbstractValue)) return this.$Get(P, Receiver);
-    invariant(this === Receiver, "TODO #1021");
     if (this.values.isTop()) {
-      if (this.isSimpleObject() && this.isIntrinsic()) {
+      if ((this.isSimpleObject() && this.isIntrinsic()) || this.$Realm.isInPureScope()) {
         return AbstractValue.createTemporalFromBuildFunction(this.$Realm, Value, [this, P], ([o, p]) =>
           t.memberExpression(o, p, true)
         );
       }
       AbstractValue.reportIntrospectionError(this);
       throw new FatalError();
     }
+    invariant(this === Receiver, "TODO #1021");
```

Comment:
I moved this below as this was another common invariant that was triggered when in combination with the pure wrapper.

Reply:
The above code is not valid if the receiver is not the same as this. At least not unless we’re sure it’s not going to be a getter.

Reply:
Currently we can’t really write tests against this code path because abstract object values can’t be the prototype of an object. I have an open wip pr to fix that so that we can properly tests this.

However the reason this fires in your case is because the receiver could be a primitive that was coerced with ToObject and therefore they’re different.

We could perhaps make a special case for that scenario which is slightly different than accessing through a prototype chain which is the case I was worried about above.

Reply:
I'm only catering for `ToObject`. I'll special case that for this.